### ISSUE TITLE
Dimentionality Reduction before machine learning

### ISSUE THREAD

**Issue by dustinmichels** [nachtm/cosmic-rAI#4]
It looks like [in some cases](https://towardsdatascience.com/dimensionality-reduction-does-pca-really-improve-classification-outcome-6e9ba21f0a32), "dimensionality reduction" via techniques like PCA can improve the performance of classifiers.

Sklearn has a suite of tools for this: http://scikit-learn.org/stable/modules/decomposition.html#decompositions

I think the "cropping" of sensors that Micah has discussed,  (to omit sensors that aren't hit for a given event) is basically also dimentionality reduction.

**Comment by simonbilskyrollins** [comment]
I'm looking at the various tools in Sklearn's `feature_selection` module - they should be able to help with this problem as well.

**Comment by dustinmichels** [comment]
I tried PCA to reduce the 100+ sensors down to n=3 or n=10 attributes, which are supposed to reflect the n-axis that explain the most variance. But they don't seem to explain much variance. I think we have to be a little careful about choosing techniques that work well with sparse data, if we wanna reduce charge features. What are you thinking about?

**Comment by dustinmichels** [comment]
I tried basically sorting the sensors for each event and picking the top 10 or 20. I was thinking this could be a naive form of cropping-- we lose the true structure of the sensors, but maybe there is still a distinctive pattern. Bad results.
